pkg/queue/metrics.go

```diff
@@ -32,15 +67,19 @@ const (
 var (
 	// largest bucket is 5 seconds
 	durationMsBuckets = []float64{10, 50, 100, 200, 300, 500, 1000, 2000, 3000, 5000}
-	mutex             = &sync.Mutex{}
-	processName       = filepath.Base(os.Args[0])
-	constLabels       = prometheus.Labels{
+	// largest bucket is 83 min
+	durationSBuckets = []float64{1, 10, 50, 100, 200, 300, 500, 1000, 2000, 3000, 5000}
+	mutex            = &sync.Mutex{}
+	processName      = filepath.Base(os.Args[0])
```

Comment:
I think go 1.8 introduced `os.Executable()`. Nothing is wrong with `os.Args[0]`, so we can keep it like this, but at least theoretically some flag parsing library could come to the idea to process the os.Args in a shifting style (like one would do in a bash script) so that after the flag parsing `os.Args` might be left empty because its just a global variable and not a constant. In this case `os.Args[0]` would produce a panic.

Reply:
this is how `os.Executable()` works:

```go
var executablePath, executablePathErr = func() (string, error) {                                                                                                                                                                                                                                                                                                                              
  var procfn string                                                                                                                                                                                                                                                                                                                                                                           
  switch runtime.GOOS {                                                                                                                                                                                                                                                                                                                                                                       
  default:                                                                                                                                                                                                                                                                                                                                                                                    
    return "", errors.New("Executable not implemented for " + runtime.GOOS)                                                                                                                                                                                                                                                                                                                   
  case "linux", "android":                                                                                                                                                                                                                                                                                                                                                                    
    procfn = "/proc/self/exe"                                                                                                                                                                                                                                                                                                                                                                 
  case "netbsd":                                                                                                                                                                                                                                                                                                                                                                              
    procfn = "/proc/curproc/exe"                                                                                                                                                                                                                                                                                                                                                              
  case "dragonfly":                                                                                                                                                                                                                                                                                                                                                                           
    procfn = "/proc/curproc/file"                                                                                                                                                                                                                                                                                                                                                             
  }                                                                                                                                                                                                                                                                                                                                                                                           
  return Readlink(procfn)                                                                                                                                                                                                                                                                                                                                                                     
}()                                                                                                                                                                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                                                                                                                              
func executable() (string, error) {                                                                                                                                                                                                                                                                                                                                                           
  return executablePath, executablePathErr                                                                                                                                                                                                                                                                                                                                                    
}
```

So, it returns an error which we have to handle and makes the link resolution in runtime which makes it also unreliable.

I think panic if the args are empty is totally fine, we will find very quick about it when we try to start the process. I've never seen a library that could edit the `os.Args` slice. I'd rather avoid handling the error here and triggering the panic by hand.

Reply:
right, error handling would overly complicate this setup for this constructed case. So lets leave it like this